### PR TITLE
Redact extra fields for Asset Endpoints in fastAPI

### PR DESCRIPTION
**Pull request by vatsrahul1001**
This PR address the review [comment](https://github.com/apache/airflow/pull/43984#discussion_r1843856417) in Asset migration [PR](https://github.com/apache/airflow/pull/43984). We noticed `extra` are redacted in the Log table, but not in the plain response from Asset Legacy API. Querying the events shouldn't return sensitive values in plain text. This PR handles that in fastAPI.

### Testing

**Asset events reponse**
image

**Asset response**

image

**create Asset response**
image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by vatsrahul1001** [comment]
@pierrejeambrun I have `redacted` for `create Asset response` as well let me know if its not needed I can remove that

File: tests/api_fastapi/core_api/routes/public/test_assets.py [hunk 1]
```diff
@@ -424,6 +475,68 @@ def test_limit_and_offset(self, test_client, params, expected_asset_uris):
         asset_uris = [asset["uri"] for asset in response.json()["asset_events"]]
         assert asset_uris == expected_asset_uris
 
+    @pytest.mark.usefixtures("time_freezer")
+    @pytest.mark.enable_redact
+    def test_should_mask_sensitive_extra_logs(self, test_client, session):
```
Comment (pierrejeambrun) [inline tests/api_fastapi/core_api/routes/public/test_assets.py]: Not sure why we have `logs` in the name of the test, I don't think we assert any `Log` here.

File: tests/api_fastapi/core_api/routes/public/test_assets.py [hunk 2]
```diff
@@ -463,17 +576,29 @@ def test_should_respond_404(self, test_client):
         assert response.status_code == 404
         assert response.json()["detail"] == "The Asset with uri: `s3://bucket/key` was not found"
 
+    @pytest.mark.usefixtures("time_freezer")
+    @pytest.mark.enable_redact
+    def test_should_mask_sensitive_extra_logs(self, test_client, session):
```
Comment (pierrejeambrun) [inline tests/api_fastapi/core_api/routes/public/test_assets.py]: same

**Review by pierrejeambrun — APPROVED** [review]
Nice

File: tests/api_fastapi/core_api/routes/public/test_assets.py (hunk 1 above)
Comment (vatsrahul1001) [reply tests/api_fastapi/core_api/routes/public/test_assets.py]: done

File: tests/api_fastapi/core_api/routes/public/test_assets.py (hunk 2 above)
Comment (vatsrahul1001) [reply tests/api_fastapi/core_api/routes/public/test_assets.py]: done

**Pull request merged**